### ISSUE TITLE
absurd_extreme_comparisons false positive for custom types

### ISSUE THREAD

**Issue by tspiteri** [Manishearth/rust-clippy#1387]
If the lhs of the comparison is a custom type for which there is a custom `PartialCmp<T>`, there can be a false positive. In the example below, `val` can be `> u32::MAX` but clippy thinks it can't.

```rust
use std::cmp::{Ordering, PartialEq, PartialOrd};

#[derive(PartialEq, PartialOrd)]
pub struct U(u64);

impl PartialEq<u32> for U {
    fn eq(&self, other: &u32) -> bool {
        self.eq(&U(*other as u64))
    }
}
impl PartialOrd<u32> for U {
    fn partial_cmp(&self, other: &u32) -> Option<Ordering> {
        self.partial_cmp(&U(*other as u64))
    }
}

pub fn foo(val: U) -> bool {
    val > std::u32::MAX
}
```
